pages/en/lb4/Defining-the-API-using-bottom-up-approach.md

```diff
@@ -0,0 +1,156 @@
+---
+lang: en
+title: 'Defining the API using bottom up approach'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Defining-the-API-using-bottom-up-approach.html
+summary:
+---
+
+## Define the API from bottom to top
+
+### Start with LoopBack artfiacts
+
+Sometimes, it's hard to know what your API is going to look like without
+writing out your application first. It also might be difficult to adjust your
+API as plans change or as features need to be reworked.
+
+With TypeScript's [experimental decorator](https://www.typescriptlang.org/docs/handbook/decorators.html)
+feature, APIs can be automatically built and exposed as your application
+continues development. Some key concepts utilize decorators to gather
+'metadata' about your code and then assemble them into a valid OpenAPI spec.
+These concepts and their decorators include:
+
+- [Model](Model.html)
+  - `@model()`
+  - `@property()`
+- [Routes](Routes.html)
+  - `@operation()`
+  - `@param()`
+
+### Define your models
+
+Your models will be used to provide schemas to validate against the data
+your API will intercept and consequently are likely to be referenced as
+argument types in your controllers, so they should be the first to be defined.
```

Comment:
- Your models are not necessarily going to be used to validate anything. I can easily write my controller functions to accept objects that neither strictly nor logically equivalent to my model objects.
- The _value_ of using models is that they give you a common, workable definition for:
  - representing inbound and outbound data at the API layer
  - handling data and relationships at the datasource layer
- _Intercept_ carries a very particular connotation; that the intended recipient is not the first to receive or act upon the incoming payload. In our case, we're not intercepting the request, we're handling it.